monks-little-details.js

```diff
@@ -703,44 +703,96 @@ background-color: rgba(0, 0, 0, 0.5);
-                } else {
-                    let combatantxp = combatant?.actor.system.details?.xp?.value;
-                    if (combatantxp == undefined) {
+        if (game.system.id == 'pf2e') {
+            //cr will just be a -1 to 3 value (representing trivial - extreme)
+            //apl will not be passed forward, instead XP is passed forward
+
+            //note, should be referenced by xpByRelLevel[relLevel + 4]
+            var xpByRelLevel = [10, 15, 20, 30, 40, 60, 80, 120, 160];
+
+            //note that this needs to be multiplied by party size
+            var crsByXp = [10, 15, 20, 30, 40]
+            
+            //determine APL, and modifiers if necessary
+            for (let combatant of combat.combatants) {
+                if (combatant.actor != undefined && combatant.token != undefined) {
+                    if (combatant.token.disposition == 1) {
+                        apl.count = apl.count + 1;
                         let levels = 0;
-                        if (combatant?.actor.system?.classes && Object.entities(combatant.actor.system?.classes).length)
-                            levels = combatant.actor.system?.classes?.reduce(c => { return c.levels; });
-                        else if (combatant?.actor.system.details?.level?.value)
-                            levels = parseInt(combatant?.actor.system.details?.level?.value);
-                        combatantxp = MonksLittleDetails.xpchart[Math.clamped(levels, 0, MonksLittleDetails.xpchart.length - 1)];
+                        if (combatant.actor.system?.classes) {
+                            levels = Object.values(combatant.actor.system?.classes).reduce((a, b) => {
+                                return a + (b?.levels || b?.level || 0);
+                            }, 0);
+                        } else {
+                            levels = combatant?.actor.system.details?.level?.value || combatant?.actor.system.details?.level || 0;
+                        }
+    
+                        apl.levels += levels;
                     }
-                    xp += (combatantxp || 0);
                 }
             }
-        };
 
-        var calcAPL = 0;
-        if (apl.count > 0)
-            calcAPL = Math.round(apl.levels / apl.count) + (apl.count < 4 ? -1 : (apl.count > 5 ? 1 : 0));
+            var calcAPL = 0;
+            if (apl.count > 0)
+                calcAPL = Math.round(apl.levels / apl.count);
+                //this approximation is fine -- most pf2e parties should all be the same level, but otherwise we can just round
+
+            //for each enemy, determine its xp value
+            for (let combatant of combat.combatants) {
+                if (combatant.actor != undefined && combatant.token != undefined) {
+                    if (combatant.token.disposition != 1) {
+                        var level = 0;
+                        level = parseInt(combatant?.actor.system.details?.level?.value);

```

Comment:
If the level value doesn't exist, then it will end up being null, and parseInt will return a NaN, which will make for an incorrect total.  There should be a `?? 0` after the value.